src/main/scala/scala/tools/refactoring/analysis/IsNotInImports.scala

```diff
@@ -0,0 +1,53 @@
+package scala.tools.refactoring
+package analysis
+
+trait IsNotInImports { _: CompilationUnitDependencies with common.EnrichedTrees =>
```

Comment:
@wpopielarski yes, the thing with path dependent types was what I meant with `because the compiler API forces their cakes on us`. Also, I don't think that stackable traits should be avoided at all costs, I just think that they should be used more cautiously and only if other alternatives are are inferior. That being said, as for `IsNotInImports`, I would have have left these methods in `CompilationUnitDependencies` without creating a new type at all, since I don't see why splitting this out makes `CompilationUnitDependencies` more readable, but maybe this is a matter of taste.